File: praca-inzynierska-frontend/src/app/app/model/TimePeriod.ts
import { TimePeriodDto } from './dto/TimePeriodDto';

export class TimePeriod {
    constructor(
        public start: string,
        public finish: string
    ) {}

    static of(timePeriod: TimePeriodDto): TimePeriod {
        if (timePeriod) {
            return new TimePeriod(timePeriod.start.toString(), timePeriod.finish.toString())
        } else {
            return null
        }
    }
}

File: praca-inzynierska-backend/application/src/main/kotlin/jwer/backend/model/embeddable/TimePeriod.kt
package jwer.backend.model.embeddable

import java.time.LocalDate
import java.time.LocalTime
import javax.persistence.Embeddable

@Embeddable
data class TimePeriod(
        var start: LocalTime,
        var finish: LocalTime
) {

    fun atDate(date: LocalDate): DateTimePeriod {
        return DateTimePeriod(
                start.atDate(date),
                finish.atDate(date)
        )
    }

}